### PR TITLE
Group results by weather

### LINKED ISSUE #44: Trip Index Page - Group Results by Weather
As an authorized user, when I visit the trip index page, I see trip results grouped into weather categories.

- [ ] Happy/sad path testing

- [ ] Update to README (FE screenshots, BE API endpoint documented)

### LINKED ISSUE #57: Go Lucky flight not caching (on localhost)
When the user clicks go Lucky, the index action correctly caches the single flight, but then redirects to the show action.  In the show action, we attempt retrieve the cached flight, but it's no longer there so returns a not_found page.

### PR DESCRIPTION
**Pull request by JesseMellinger**
Closes #44 
Closes #57 

- [x] Updated Files:
  `app/controllers/flights_controller.rb`
  `app/poros/trip.rb`
  `app/views/flights/_flight_partial.html.erb`
  `app/views/flights/index.html.erb`
  `spec/features/flights/flight_information_service_spec.rb`
  `spec/poros/trip_spec.rb`
  `spec/fixtures/flight_data_return.json`

- [x] Repo Representation: Front End, Back End, Kiwi, Open Weather, Google 
  Front End

- [ ] Sad Path Tested

- [x] Happy Path Tested

- [x] All Tests are passing

- [x] Collaborators:
  Sage Gonzales
  Jesse Mellinger

### CODE REVIEW

**Review by stillsheryl — APPROVED** [review]
Great use of a partial, and I love the fun descriptions! Way to give our app some character! :)

File: app/controllers/flights_controller.rb [hunk 1]
```diff
@@ -8,6 +8,7 @@ def index
       @trips = SearchFacade.get_flights(flight_params)
       return error_flash(@trips) if @trips.class == String
       write_to_cache(@trips)
+      @hot_trips, @warm_trips, @cool_trips, @cold_trips = Trip.group_by_weather(@trips)
```
Comment (GeorgieGirl24) [inline app/controllers/flights_controller.rb:11]: awesome!

File: app/poros/trip.rb [hunk 2]
```diff
@@ -40,6 +40,19 @@ def initialize(info)
     @booking_link = info[:attributes][:booking_link]
     @description = info[:attributes][:description]
   end
+  
+  class << self 
+    def group_by_weather(trips)
+      hot_trips, warm_trips, cool_trips, cold_trips = [], [], [], []
+      trips.each do |trip|
+        hot_trips << trip if trip.day_feels_like_f >= 85
+        warm_trips << trip if trip.day_feels_like_f >= 70 && trip.day_feels_like_f < 85
```
Comment (GeorgieGirl24) [inline app/poros/trip.rb:49]: Great job with four different kinds of groupings

File: app/views/flights/index.html.erb [hunk 3]
```diff
@@ -1,20 +1,31 @@
...
-      Feels Like: <%= trip.day_feels_like_f.round(1) %>, <br>
-      Weather Description: <%= trip.description %>, <br>
-  </tr>
-  <% end %>
-</table>
+
+<% if !@hot_trips.empty? %>
+<h2>Bring Suntan Lotion (85° and up)</h2>
+  <section class="hot-trips">
+    <%= render partial: 'flight_partial', collection: @hot_trips, as: :trip %>
```
Comment (GeorgieGirl24) [inline app/views/flights/index.html.erb:6]: great use of partials

**Review by GeorgieGirl24 — APPROVED** [review]

**Review by cpfergus1 — APPROVED** [review]
Good deal, I am looking forward to seeing it function!

**Pull request merged**